game-app/game-core/src/main/java/games/strategy/triplea/TripleAPlayer.java

```diff
@@ -256,6 +255,14 @@ public boolean acceptAction(
             politics);
   }
 
+  private void playSound(final String soundPath) {
+    if (ui == null) {
+      log.warn("UI not set when trying to play sound");
```

Comment:
The warning will show a pop up to user, seemingly this will not be very informative to them.
Are we sure that `ui` will possibly be null at this point? If so, perhaps it is null for a valid reasons - in those cases, should we be showing a warning message to the user? 
I wonder a bit as well if `TripleAPlayer` should have a function for playing sounds which we could bind to the uiContext clip player when the uiContext is set.

Reply:
This warning should not happen under normal circumstances. `HeadedLaunchAction#startGame` sets the frame on `TripleAPlayer`. Thinking about it this really should be either an exception that is thrown or an error that gets logged instead of a warning, because ui can only ever be null if the code tries to play a sound before the game is completely initialized. `TripleAPlayer` only ever plays sound based on gameplay events if I understand this correctly.